src/components/promotion/PromotionContainer.js

```diff
@@ -0,0 +1,210 @@
+import useGlobalStore from 'store/GlobalStore';
+import PromoteCard from 'components/promotion/PromoteCard/PromoteCard';
+
+configure({
+  axios: axiosInstance,
+});
+
+const { Content } = Layout;
+
+export default function () {
+  const [searchString, setSearchString] = useState('');
+  const [showPromoted, setShowPromoted] = useState(true);
+  const [filters, setFilters] = useState({
+    offer: true,
+    pathway: true,
+    provider: true,
+  });
+  const {
+    provider: providerStore,
+    pathway: pathwayStore,
+    offer: offerStore,
+  } = useGlobalStore();
+
+  const [{ data: getPathways }] = useAxios('/pathways?scope=with_files');
+
+  const [{ data: getOffers }] = useAxios('/offers?scope=with_files');
+
+  const [{ data: getProviders }] = useAxios('/providers?scope=with_files');
+
+  const offers = Object.values(offerStore.entities).map((o) => {
+    return {
+      ...o,
+      entity_type: 'offer',
+    };
+  });
+
+  const providers = Object.values(providerStore.entities).map((p) => {
+    return {
+      ...p,
+      entity_type: 'provider',
+    };
+  });
+
+  const pathways = Object.values(pathwayStore.entities).map((p) => {
+    return {
+      ...p,
+      entity_type: 'pathway',
+    };
+  });
+
+  const data = [...offers, ...pathways, ...providers];
+
+  function toggleFilter(key = 'offer') {
+    setFilters({
+      ...filters,
+      [key]: !filters[key],
+    });
+  }
+
+  function filterContent() {
+    return (
+      <div>
+        <Row className="justify-between mb-1" gutter={6}>
+          <Col>Providers</Col>
+          <Col>
+            <Switch
+              size="small"
+              defaultChecked={filters.provider}
+              onClick={() => toggleFilter('provider')}
+            />
+          </Col>
+        </Row>
+        <Row className="justify-between mb-1" gutter={6}>
+          <Col>Offers</Col>
+          <Col>
+            <Switch
+              size="small"
+              defaultChecked={filters.offer}
+              onClick={() => toggleFilter('offer')}
+            />
+          </Col>
+        </Row>
+        <Row className="justify-between" gutter={6}>
+          <Col>Pathways</Col>
+          <Col>
+            <Switch
+              size="small"
+              defaultChecked={filters.pathway}
+              onClick={() => toggleFilter('pathway')}
+            />
+          </Col>
+        </Row>
+      </div>
+    );
+  }
+
+  const handleDataSearch = (searchVal) => {
+    return setSearchString(searchVal);
+  };
+
+  const handleOnChange = (event) => {
+    if (showPromoted) {
+      setShowPromoted(false);
+    }
+    if (event.target.value.length === 0) {
+      setShowPromoted(true);
+    }
+    setSearchString(event.target.value);
+  };
+
+  const handleDataAfterSearch = (data, keys = ['name']) => {
+    return matchSorter(data, searchString, { keys });
+  };
+
+  let showData = handleDataAfterSearch(data);
+
+  if (showPromoted) {
+    showData = showData.filter((d) => {
+      if (

```

Comment:
im unsure what is the data look like but this can be simplify for sure, can you paste date format